### PR TITLE
[Feature][Core][Connector-V2] Unified The way of setting JobName

### LINKED ISSUE #2809: [Feature][Core][Connector-V2] Unified The way of setting JobName
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

In connector-v2, flink can set job name use `job.name` in `env` in config. But spark can't set job name(application name) at now.
Our goal:
Both flink and spark:
1. User can use same way to set job name in command.
2. User can use same way to set job name in `env` in config.

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Carl-Zhou-CN**
## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

**Comment by Carl-Zhou-CN** [comment]
#2809

**Comment by Carl-Zhou-CN** [comment]
**spark**
  ${SPARK_HOME}/bin/spark-submit --class "org.apache.seatunnel.core.starter.spark.SeatunnelSpark" --name "test1" --master "local[4]" --deploy-mode "client" --conf "spark.executor.memory=1g" --conf "spark.streaming.batchDuration=5" --conf "spark.executor.cores=1" --conf "job.name=SeaTunnel" --conf "spark.executor.instances=2" /data/seatunnel-2.1.3/lib/seatunnel-spark-starter.jar -n test1 --master local[4] --deploy-mode client --config ./config/spark.streaming.conf.template
**flink**
 ${FLINK_HOME}/bin/flink run -c org.apache.seatunnel.core.starter.flink.SeatunnelFlink /data/seatunnel-2.1.3/lib/seatunnel-flink-starter.jar --config ./config/flink.streaming.conf.template -Dpipeline.name=test

Env takes precedence over the command line,job.name is used to set the application name

File: seatunnel-core/seatunnel-core-starter/src/main/java/org/apache/seatunnel/core/starter/command/AbstractCommandArgs.java [hunk 1]
```diff
@@ -41,6 +42,10 @@ public abstract class AbstractCommandArgs implements CommandArgs {
             description = "check config")
     private boolean checkConfig = false;
 
+    @Parameter(names = {"-n", "--name"},
```
Comment (EricJoy2048) [inline seatunnel-core/seatunnel-core-starter/src/main/java/org/apache/seatunnel/core/starter/command/AbstractCommandArgs.java:45]: Please add the new parameter to the document https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/command/usage.mdx

Comment (Carl-Zhou-CN) [reply seatunnel-core/seatunnel-core-starter/src/main/java/org/apache/seatunnel/core/starter/command/AbstractCommandArgs.java:45]: ok

Comment (Carl-Zhou-CN) [reply seatunnel-core/seatunnel-core-starter/src/main/java/org/apache/seatunnel/core/starter/command/AbstractCommandArgs.java:45]: @EricJoy2048 Currently, this parameter is only supported for V2. Is there a special document for V2

Comment (EricJoy2048) [reply seatunnel-core/seatunnel-core-starter/src/main/java/org/apache/seatunnel/core/starter/command/AbstractCommandArgs.java:45]: > 

Can we add V2 starter in this doc https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/command/usage.mdx ?

Comment (EricJoy2048) [reply seatunnel-core/seatunnel-core-starter/src/main/java/org/apache/seatunnel/core/starter/command/AbstractCommandArgs.java:45]: > > 
> 
> Can we add V2 starter in this doc https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/command/usage.mdx ? @CalvinKirs

Comment (CalvinKirs) [reply seatunnel-core/seatunnel-core-starter/src/main/java/org/apache/seatunnel/core/starter/command/AbstractCommandArgs.java:45]: in fact, `connectot-v2` it's not much different from v1, it's just the scripts and jars that are started. Are you willing to submit documents in this regard?

Comment (Carl-Zhou-CN) [reply seatunnel-core/seatunnel-core-starter/src/main/java/org/apache/seatunnel/core/starter/command/AbstractCommandArgs.java:45]: > > 
> 
> Can we add V2 starter in this doc https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/command/usage.mdx ?

If it's like EricJoy2048 says, I can try

Comment (CalvinKirs) [reply seatunnel-core/seatunnel-core-starter/src/main/java/org/apache/seatunnel/core/starter/command/AbstractCommandArgs.java:45]: if you have any question, you can ping me

**Review by EricJoy2048 — DISMISSED** [review]
LGTM

**Comment by Carl-Zhou-CN** [comment]
@EricJoy2048 I don't seem to have added or changed any dependencies

**Comment by EricJoy2048** [comment]
> @EricJoy2048 I don't seem to have added or changed any dependencies

Because the `st-engine` branch merged to `dev` and the licenses have some problems. So please wait the pr https://github.com/apache/incubator-seatunnel/pull/2957 merged. And then I will rerun the ci of your pr.

**Comment by Carl-Zhou-CN** [comment]
@EricJoy2048 Please help me run the ci again

**Review by CalvinKirs — DISMISSED** [review]

**Comment by Carl-Zhou-CN** [comment]
@EricJoy2048 @CalvinKirs Please help review again, there are some conflicts in merge's new code

**Review by hailin0 — COMMENTED** [review]
LGTM

**Review by CalvinKirs — APPROVED** [review]

**Review by EricJoy2048 — APPROVED** [review]
+1

**Pull request merged**